packages/deployments/contracts/contracts/core/connext/helpers/RelayerProxy.sol

```diff
@@ -300,6 +354,36 @@ contract RelayerProxy is ProposedOwnable, ReentrancyGuard, GelatoRelayFeeCollect
     transferRelayerFee(_relayerFee);
   }
 
+  /**
+   * @notice Wraps the call to proposeAggregateRoot() on SpokeConnector and pays either the caller or hardcoded relayer
+   * @dev _rootTimestamp is required for off-chain agents to be able to know which root they should fetch from the root manager contract
+   *                     in order to compare it with the one being proposed. The off-chain agents should also ensure the proposed root is
+   *                     not an old one.
+   * @param _aggregateRoot The aggregate root to propose.
+   * @param _rootTimestamp Block.timestamp at which the root was finalized in the root manager contract.
+   * @param _signature Signature from the approved proposer.
+   * @param _fee - Fee to be paid to relayer.
+   */
+  function proposeAggregateRoot(
+    bytes32 _aggregateRoot,
+    uint256 _rootTimestamp,
+    bytes memory _signature,
+    uint256 _fee
+  ) external onlyRelayer nonReentrant {
+    if (!_proposeAggregateRootCooledDown()) {
+      revert RelayerProxy__proposeAggregateRootCooledDown_notCooledDown();
+    }
+
+    // Validate the signer
+    _validateProposeSignature(_aggregateRoot, _rootTimestamp, lastProposeAggregateRootAt, nonce++, _signature);
+
```

Comment:
Add chainId as well?

Reply:
Why don't we use `lastProposeAggregateRootAt` in  `_validateProposeSignature` function directly? I believe that will save some gas?

Reply:
@liu-zhipeng adding lastProposeAggregateRootAt, is sufficient right ? Don't need nonce as well.

Reply:
`lastProposeAggregateRootAt` might be insufficient. because malicious actor can propose with same signature again and again.

Reply:
@preethamr  you are correct, we increased `lastProposeAggregateRootAt ` after proposing. So we can use that as the nonce.

Reply:
Added chain id